Tests/BackendIntegrationTests/OfflineStoreKitIntegrationTests.swift

```diff
@@ -246,6 +245,37 @@ class OfflineStoreKit1IntegrationTests: BaseOfflineStoreKitIntegrationTests {
         )
     }
 
+    @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+    func testCallToGetCustomerInfoWithPendingTransactionsPostsReceiptOnlyOnce() async throws {
+        self.serverDown()
+
+        try await self.purchaseMonthlyProduct()
+        try self.testSession.forceRenewalOfSubscription(
+            productIdentifier: await self.monthlyPackage.storeProduct.productIdentifier
+        )
+
+        try await asyncWait(description: "Expected 2 unfinished transactions") {
+            await Transaction.unfinished.extractValues().count == 2
+        }
+
+        self.serverUp()
+
+        let customerInfo = try await self.purchases.customerInfo(fetchPolicy: .fetchCurrent)
+        try await self.verifyEntitlementWentThrough(customerInfo)
+
+        self.logger.verifyMessageWasLogged(
+            "Found 2 unfinished transactions, will post receipt in lieu of fetching CustomerInfo",
+            level: .debug,
+            expectedCount: 1
+        )
+
+        self.logger.verifyMessageWasLogged(
```

Comment:
This is non deterministically failing due to 2 different receipt hashes:
```
PostReceiptDataOperation $RCAnonymousID:0ebfbac4bab74d49ba12148425978aa8-true-3fca657f1089ffede6395b3143f2fc583582dc5163f414229bd860b179def565
-com.revenuecat.monthly_4.99.1_week_intro-4.99-USD-USA-2-0-7096FF06-P1M-P1W-2-com.revenuecat.monthly_4.99.1_free_week
--false
-["$attConsentStatus": [SubscriberAttribute] key: $attConsentStatus value: notDetermined setTime: 2023-08-02 22:03:24 +0000]'
```

```
PostReceiptDataOperation $RCAnonymousID:0ebfbac4bab74d49ba12148425978aa8-false-f68c17b6aec2126857f2f64ba4e7e4221cfd14c6e3ffa5f36f8b32e94205671e
-com.revenuecat.monthly_4.99.1_week_intro-4.99-USD-USA-2-0-7096FF06-P1M-P1W-2-com.revenuecat.monthly_4.99.1_free_week
--false
-["$attConsentStatus": [SubscriberAttribute] key: $attConsentStatus value: notDetermined setTime: 2023-08-02 22:03:25 +0000]'
```

In the logs you can see it's because it's parsing the receipt twice.